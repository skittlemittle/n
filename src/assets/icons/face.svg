<svg id="svg" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" width="400" height="415.31449407474935" viewBox="0, 0, 400,415.31449407474935"><g id="svgg"><path id="path0" d="M120.860 72.518 C 116.889 74.322,115.470 79.359,117.999 82.675 C 119.153 84.188,119.346 85.241,119.920 93.163 C 120.109 95.770,120.620 99.380,121.055 101.185 C 121.841 104.442,121.847 104.732,121.807 139.289 C 121.774 168.444,121.857 174.823,122.321 178.487 C 122.787 182.173,122.875 189.458,122.878 224.718 C 122.880 265.186,122.904 266.619,123.576 267.927 C 126.036 272.708,131.436 273.471,134.984 269.540 C 136.873 267.446,136.788 270.025,136.575 221.696 C 136.146 124.477,135.859 102.296,134.982 98.815 C 134.553 97.110,134.197 94.649,134.191 93.345 C 134.185 92.042,134.014 89.991,133.812 88.788 C 133.609 87.584,133.374 84.991,133.290 83.025 C 132.950 75.122,126.720 69.856,120.860 72.518 M293.530 80.779 C 291.861 81.530,290.700 82.642,289.846 84.306 C 289.170 85.624,289.152 87.751,289.152 166.268 L 289.152 246.877 288.241 249.043 C 286.948 252.113,286.817 269.945,288.068 272.399 C 290.478 277.122,297.047 277.497,299.973 273.077 L 301.003 271.523 301.111 263.109 C 301.214 255.018,301.253 254.625,302.113 252.891 L 303.008 251.087 303.008 168.445 L 303.008 85.803 302.160 84.196 C 300.467 80.990,296.530 79.431,293.530 80.779 M205.469 331.624 C 199.077 333.924,199.384 343.105,205.915 344.919 C 207.808 345.444,231.249 345.431,233.181 344.903 C 239.150 343.273,240.353 335.968,235.244 332.376 L 233.667 331.267 220.297 331.183 C 209.425 331.115,206.656 331.197,205.469 331.624 " stroke="none" fill="#050505" fill-rule="evenodd"></path><path id="path1" d="M0.000 207.657 L 0.000 415.314 200.000 415.314 L 400.000 415.314 400.000 207.657 L 400.000 0.000 200.000 0.000 L 0.000 0.000 0.000 207.657 M127.812 72.795 C 129.854 73.762,131.229 75.199,132.647 77.849 C 133.021 78.547,133.339 80.493,133.526 83.227 C 134.267 94.049,134.506 96.314,135.265 99.727 C 136.074 103.365,136.076 103.447,136.033 138.560 C 135.996 168.048,136.078 174.396,136.538 177.758 C 137.000 181.135,137.088 188.454,137.094 224.085 L 137.101 266.402 136.154 268.302 C 134.030 272.565,128.775 273.602,125.201 270.463 C 122.497 268.090,122.729 272.054,122.427 222.972 C 121.733 109.896,121.649 103.705,120.773 100.273 C 120.305 98.439,119.973 95.984,119.960 94.257 C 119.948 92.653,119.703 90.273,119.417 88.970 C 119.130 87.666,118.889 86.154,118.882 85.610 C 118.875 85.065,118.390 83.863,117.803 82.938 C 113.408 76.008,120.156 69.171,127.812 72.795 M299.283 80.857 C 301.019 81.656,302.561 83.717,303.018 85.850 C 303.297 87.151,303.380 111.771,303.302 169.553 L 303.191 251.413 302.370 252.948 C 301.595 254.398,301.550 254.903,301.548 262.064 C 301.546 266.406,301.375 270.268,301.147 271.103 C 299.334 277.741,290.189 278.296,287.529 271.930 C 286.443 269.332,286.717 251.440,287.876 249.260 L 288.788 247.546 288.796 166.708 C 288.803 88.802,288.828 85.818,289.479 84.412 C 291.265 80.554,295.382 79.061,299.283 80.857 M234.746 331.779 C 236.456 332.656,237.551 333.826,238.269 335.545 C 239.705 338.983,238.208 342.836,234.746 344.612 C 233.160 345.425,207.095 345.694,205.037 344.918 C 199.270 342.744,199.077 334.091,204.740 331.595 C 206.714 330.725,233.005 330.886,234.746 331.779 " stroke="none" fill="#fbfbfb" fill-rule="evenodd"></path><path id="path2" d="M122.972 71.719 C 123.323 71.811,123.897 71.811,124.248 71.719 C 124.599 71.627,124.312 71.552,123.610 71.552 C 122.908 71.552,122.621 71.627,122.972 71.719 M118.962 85.871 C 118.966 86.673,119.040 86.958,119.128 86.504 C 119.215 86.051,119.211 85.395,119.120 85.046 C 119.029 84.697,118.958 85.068,118.962 85.871 M303.187 168.459 C 303.187 213.683,303.230 232.183,303.282 209.572 C 303.334 186.960,303.334 149.959,303.282 127.347 C 303.230 104.736,303.187 123.236,303.187 168.459 M136.099 108.478 C 136.103 109.280,136.178 109.565,136.265 109.112 C 136.352 108.658,136.349 108.002,136.258 107.653 C 136.167 107.304,136.095 107.675,136.099 108.478 M301.340 262.352 C 301.341 266.764,301.393 268.519,301.457 266.251 C 301.522 263.984,301.521 260.374,301.457 258.230 C 301.393 256.085,301.340 257.940,301.340 262.352 " stroke="none" fill="#a4a4a4" fill-rule="evenodd"></path><path id="path3" d="M122.692 223.883 C 122.692 246.846,122.736 256.240,122.789 244.758 C 122.843 233.277,122.843 214.490,122.789 203.008 C 122.736 191.527,122.692 200.921,122.692 223.883 " stroke="none" fill="#8c8c8c" fill-rule="evenodd"></path><path id="path4" d="M288.967 166.819 C 288.967 211.139,289.009 229.325,289.061 207.231 C 289.113 185.136,289.113 148.874,289.061 126.647 C 289.009 104.421,288.967 122.498,288.967 166.819 M121.903 109.754 C 121.903 111.057,121.970 111.591,122.050 110.939 C 122.131 110.287,122.131 109.221,122.050 108.569 C 121.970 107.917,121.903 108.450,121.903 109.754 M136.913 223.519 C 136.913 246.682,136.957 256.158,137.010 244.576 C 137.063 232.995,137.063 214.043,137.010 202.461 C 136.957 190.880,136.913 200.356,136.913 223.519 M287.123 260.893 C 287.124 266.108,287.175 268.190,287.237 265.521 C 287.300 262.852,287.299 258.586,287.237 256.040 C 287.174 253.495,287.123 255.679,287.123 260.893 " stroke="none" fill="#747474" fill-rule="evenodd"></path></g></svg>