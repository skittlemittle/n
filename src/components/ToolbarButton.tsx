import styled from "styled-components";

const Tbutton = styled.button`
  height: 40px;
  width: 40px;
  background: transparent;
  border: none;
  background-image: url(${(props) => props.resource});
  background-size: cover;
  background-repeat: no-repeat;
  margin: none;
  padding: 0px;
`;

interface props {
  iconUrl: string;
  action: () => void;
}

/** image button */
const ToolbarButton = (props: props) => (
  <Tbutton onClick={props.action} resource={props.iconUrl} />
);

export default ToolbarButton;
