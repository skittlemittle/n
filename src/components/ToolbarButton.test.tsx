import renderer from "react-test-renderer";
import ToolbarButton from "./ToolbarButton";
import face from "../assets/icons/face.svg";

it("ToolbarButton renders", () => {
  const tree = renderer
    .create(<ToolbarButton iconUrl={face} action={() => {}} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
