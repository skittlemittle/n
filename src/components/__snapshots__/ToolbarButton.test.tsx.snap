// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToolbarButton renders 1`] = `
<button
  className="sc-bdvvtL kYYPCm"
  onClick={[Function]}
  resource="face.svg"
/>
`;
